test/test_nn.py

```diff
@@ -3390,7 +3390,7 @@ def _test_LayerNorm_general(self, device="cpu", dtype=torch.float):
             self.assertRaises(RuntimeError, lambda: ln(input))
 
     def _test_LayerNorm_cuda_half(self):
-        input = Variable(torch.empty(2, 3, 3, 2).to("cuda", torch.half).random_(1, 10), requires_grad=True)
+        input = torch.empty(2, 3, 3, 2).to("cuda", torch.half).random_(1, 10).requires_grad_(True)
```

Comment:
We can combine `empty` and `to` in the following manner:
`torch.empty(2, 3, 3, 2, device='cuda', dtype=torch.half)`. It's more efficient because we don't incur the host-to-device transfer.